### PR TITLE
add test cases for `<date|time> <=> <date|time>`

### PR DESCRIPTION
**Pull request by romainfrancois**
I initially went with something like: 

```cpp
set.AddFunction(
	    ScalarFunction({LogicalType::DATE, LogicalType::DATE}, LogicalType::BOOLEAN, BaseRRelopFunctionDate<OP>));
```

but it actually appears that we get it for free, so I suppose `ScalarFunctionSet` is able to find the right variant to handle dates.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+test-generated-*.R  linguist-generated=true
+relop-*.R   linguist-generated=true
```
Comment (romainfrancois) [inline .gitattributes:2]: This makes it easier to review these pull requests

**Comment by romainfrancois** [comment]
Not sure what to do with `<date> op <time>` 

They give warnings in R, and different results: 

``` r
library(duckdb)
#> Loading required package: DBI
library(duckdbrfuns)
time <- as.POSIXct(strptime('2024-02-21 14:00:00', format = '%Y-%m-%d %H:%M:%S', tz = 'UTC'))
date <- as.Date("2024-02-21")

time < date
#> Warning: Incompatible methods ("Ops.POSIXt", "Ops.Date") for "<"
#> [1] FALSE

con <- duckdbrfuns:::con()
in_df <- tibble::tibble(x1 = time, x2 = date + 1)
in_rel <- duckdb:::rel_from_df(con, in_df)
out_rel <- duckdb:::rel_project(
  in_rel,
  list(duckdb:::expr_function(
    "r_base::<",
    list(
      duckdb:::expr_reference("x1"),
      duckdb:::expr_reference("x2")
    )
  ))
)
duckdb:::rel_to_altrep(out_rel)
#>   r_base::<(x1, x2)
#> 1              TRUE
```

So this probably needs `ScalarFunction({LogicalType::DATE, LogicalType::TIME}` variants

File: duckdb-rfuns-r/tests/testthat/gen/relop.txt [hunk 2]
```diff
@@ -1,3 +1,15 @@
+"<date> <=> <date>"
+date <=> date
+date <=> date + 1
+NA_real_ <=> date
+date <=> NA_real_
```
Comment (krlmlr) [inline duckdb-rfuns-r/tests/testthat/gen/relop.txt:5]: Of course, the interesting cases are comparing dates with logicals or strings.

**Review by krlmlr — COMMENTED** [review]
Thanks, great!

**Pull request merged**